src/builtins/hmr-runtime.js

```diff
@@ -43,14 +45,53 @@ if ((!parent || !parent.isParcelRequire) && typeof WebSocket !== 'undefined') {
 
     if (data.type === 'error-resolved') {
       console.log('[parcel] ✨ Error resolved');
+
+      removeErrorOverlay();
     }
 
     if (data.type === 'error') {
       console.error('[parcel] 🚨  ' + data.error.message + '\n' + data.error.stack);
+
+      removeErrorOverlay();
+
+      var overlay = createErrorOverlay(data);
+      document.body.append(overlay);
     }
   };
 }
 
+function removeErrorOverlay() {
+  var overlay = document.getElementById(OVERLAY_ID);
+  if (overlay) {
+    overlay.remove();
+  }
+}
+
+function createErrorOverlay(data) {
+  var overlay = document.createElement('div');
+  overlay.id = OVERLAY_ID;
+
+  // html encode message and stack trace
+  var message = document.createElement('div');
+  var stackTrace = document.createElement('pre');
+  message.innerText = data.error.message
+  stackTrace.innerText = data.error.stack
+
+  var html = (
+    '<div style="background: black; font-size: 16px; color: white; position: fixed; height: 100%; width: 100%; top: 0px; left: 0px; padding: 30px; opacity: 0.85; font-family: Menlo, Consolas, monospace; z-index: 9999;">' +
+      '<span style="background: red; padding: 2px 4px; border-radius: 2px;">ERROR</span>' +
+      '<span style="top: 2px; margin-left: 5px; position: relative;">🚨</span>' +
+      '<div style="font-size: 18px; font-weight: bold; margin-top: 20px;">' + message.innerHTML + '</div>' +
+      '<pre>' + stackTrace.innerHTML + '</pre>' +
+    '</div>'
+  )
+
+  overlay.innerHTML = html;
```

Comment:
Instead of wrapping the overlay in another div, we could try using`outerHTML` instead:

```js
function createErrorOverlay(data) {
  var overlay = document.createElement('div');
  
  var html = (
    '<div id="' + OVERLAY_ID + ' ...' +
    '</div>'
  )

  overlay.outerHTML = html;

  return overlay;
}
```

Reply:
good idea will do!

Reply:
We decided to get rid of the html var as you suggested, but we kept the outer div because it just seemed more straightforward / easy to read and understand.

Reply:
👍